File: tests/analytical/plate.py
#! /usr/bin/env python

import numpy as np
import sdf_tools

offs = [0., 0., 0.]
dims = [3, 4, 5]
exts = [1.0, 2.0, 4.0]

grid = sdf_tools.Grid.Uniform(dims, offs, exts)

plate = sdf_tools.Sdf.Plate(point  = [0.1*exts[0], 0, 0],
                            normal = [1, 0, 0])

plate.apply(grid)

np.savetxt("grid.txt", grid.get())

# TEST: analytical.plate
# cd analytical
# rm -rf *.txt
# ./plate.py
# mv grid.txt grid.out.txt

File: src/sdf_tools/core/sdf/plate.cpp
#include "plate.h"

#include <sdf_tools/core/grid.h>
#include <sdf_tools/core/utils/helper_math.h>

namespace sdf_tools {
namespace sdf {

SdfPlate::SdfPlate(real3 point, real3 normal) :
    point_(point),
    normal_(normal)
{
    const real nrm = length(normal_);
    normal_ *= real(1.0) / nrm;
}

real SdfPlate::at(real3 r) const
{
    const real3 dr = r - point_;
    return dot(dr, normal_);
}

} // namespace sdf
} // namespace sdf_tools
